e2e-tests/spring-rest-mongo/src/test/java/com/foo/spring/rest/mongo/SpringRestMongoController.java

```diff
@@ -0,0 +1,115 @@
+package com.foo.spring.rest.mongo;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.ServerAddress;
+import org.evomaster.client.java.controller.EmbeddedSutController;
+import org.evomaster.client.java.controller.api.dto.AuthenticationDto;
+import org.evomaster.client.java.controller.api.dto.SutInfoDto;
+import org.evomaster.client.java.controller.problem.ProblemInfo;
+import org.evomaster.client.java.controller.problem.RestProblem;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.testcontainers.containers.GenericContainer;
+
+import java.sql.Connection;
+import java.util.List;
+import java.util.Map;
+
+public abstract class SpringRestMongoController extends EmbeddedSutController {
+
+    private final Class<?> applicationClass;
+
+    private ConfigurableApplicationContext ctx;
+    private MongoClient mongoClient;
+
+    public SpringRestMongoController(Class<?> applicationClass) {
+        this.applicationClass = applicationClass;
+    }
+
+    private final static int MONGO_PORT = 27017; //https://docs.mongodb.com/manual/reference/default-mongodb-port/
+
+    private final GenericContainer mongodb = new GenericContainer("mongo:3.2")
+            .withExposedPorts(MONGO_PORT);
+
+    @Override
+    public boolean isSutRunning() {
+        return ctx != null && ctx.isRunning();
+    }
+
+    @Override
+    public List<AuthenticationDto> getInfoForAuthentication() {
+        return null;
+    }
+
+    @Override
+    public Connection getConnection() {
+        return null;
+    }
+
+    @Override
+    public String getDatabaseDriverName() {
+        return "org.h2.Driver";
+    }
+
+    @Override
+    public ProblemInfo getProblemInfo() {
+        return new RestProblem(
+                "http://localhost:" + getSutPort() + "/v2/api-docs",
+                null
+        );
+    }
+
+    @Override
+    public SutInfoDto.OutputFormat getPreferredOutputFormat() {
+        return SutInfoDto.OutputFormat.JAVA_JUNIT_4;
+    }
+
+    @Override
+    public String startSut() {
+
+        mongodb.start();
+        ServerAddress serverAddress = new ServerAddress(mongodb.getContainerIpAddress(), mongodb.getMappedPort(MONGO_PORT));
+        MongoClientOptions mongoClientOptions = MongoClientOptions.builder()
+                .build();
+        mongoClient = new MongoClient(serverAddress, mongoClientOptions);
+
+        String host = mongodb.getContainerIpAddress();
+        int port = mongodb.getMappedPort(MONGO_PORT);
+
+        ctx = SpringApplication.run(applicationClass,
+                "--server.port=0",
+                "--spring.mongodb.embedded.version=3.2",
```

Comment:
Spring has an option to use an embedded Mongo, but that is not a real one, simply a very simplified version. 
as we use Docker to start a real Mongo, i do not think we need this setting here

Reply:
ok